File: src/stroke.ts
import { Vector } from "./vector";

export { Stroke };

/**
 * I have found some alphabet dimensions to be ~4 units in height and ~units in
 * width
 */
class Stroke {
  constructor(private start: Vector, private end: Vector) {}

  public Draw(
    ctx: CanvasRenderingContext2D,
    center: Vector,
    dimensions: Vector,
    rotation: number
  ): void {
    const halfOne = Vector.one().scale(0.5);
    const reAdjust = center.subtract(dimensions.scale(0.5));

    const startCentered = this.start.subtract(halfOne);
    const originalStartAngle = Math.atan2(startCentered.y(), startCentered.x());

    let scaledStart = new Vector(
      startCentered.magnitude() * Math.cos(rotation + originalStartAngle),
      startCentered.magnitude() * Math.sin(rotation + originalStartAngle)
    ).add(halfOne);

    scaledStart = new Vector(
      scaledStart.x() * dimensions.x(),
      scaledStart.y() * dimensions.y()
    ).add(reAdjust);

    const endCentered = this.end.subtract(halfOne);
    const originalEndAngle = Math.atan2(endCentered.y(), endCentered.x());

    let scaledEnd = new Vector(
      endCentered.magnitude() * Math.cos(rotation + originalEndAngle),
      endCentered.magnitude() * Math.sin(rotation + originalEndAngle)
    ).add(halfOne);

    scaledEnd = new Vector(
      scaledEnd.x() * dimensions.x(),
      scaledEnd.y() * dimensions.y()
    ).add(reAdjust);

    //const temp = ctx.lineWidth;
    //ctx.lineWidth = dimensions.magnitude() / 20;
    ctx.beginPath();

    ctx.lineTo(scaledStart.x(), scaledStart.y());
    ctx.lineTo(scaledEnd.x(), scaledEnd.y());
    ctx.closePath();

    ctx.stroke();
    //ctx.lineWidth = temp;
  }
}

File: UnityExample/Assets/Transmutation/Scripts/Stroke.cs
using UnityEngine;

namespace EliCDavis.Transmutation
{
    public class Stroke
    {

        private Vector2 start;

        private Vector2 end;

        public Stroke(Vector2 start, Vector2 end)
        {
            this.start = start;
            this.end = end;
        }

        public void Draw(Vector2 center, Vector2 dimensions, float rotation, IDrawingTool drawingTool)
        {
            var halfOne = Vector2.one / 2.0f;
            var reAdjust = center - (dimensions * 0.5f);

            var startCentered = start - halfOne;
            var originalStartAngle = Mathf.Atan2(startCentered.y, startCentered.x);

            var scaledStart = new Vector2(
              startCentered.magnitude * Mathf.Cos(rotation + originalStartAngle),
              startCentered.magnitude * Mathf.Sin(rotation + originalStartAngle)
            ) + halfOne;

            scaledStart = new Vector2(
              scaledStart.x * dimensions.x,
              scaledStart.y * dimensions.y
            ) + reAdjust;

            var endCentered = this.end - halfOne;
            var originalEndAngle = Mathf.Atan2(endCentered.y, endCentered.x);

            var scaledEnd = new Vector2(
              endCentered.magnitude * Mathf.Cos(rotation + originalEndAngle),
              endCentered.magnitude * Mathf.Sin(rotation + originalEndAngle)
            ) + halfOne;

            scaledEnd = new Vector2(
              scaledEnd.x * dimensions.x,
              scaledEnd.y * dimensions.y
            ) + reAdjust;

            drawingTool.Line(scaledStart, scaledEnd, dimensions.magnitude);
        }

    }

}